app/components/views/TransactionsPage/SendTab/SendTab.jsx

```diff
@@ -0,0 +1,352 @@
+  const [isSendSelf, setIsSendSelf] = useState(false);
+  const [isSendAll, setIsSendAll] = useState(false);
+  const [insuficientFunds, setInsuficientFunds] = useState(false);
+  const [showPassphraseModal, setShowPassphraseModal] = useState(false);
+
+  const prevPublishTxResponse = usePrevious(publishTxResponse);
+  const prevNextAddress = usePrevious(nextAddress);
+  const prevIsSendSelf = usePrevious(isSendSelf);
+
+  const onValidateAddress = async ({ address, index }) => {
+    let error;
+    const ref = { ...outputs[index] };
+    ref.data.destination = address;
+    if (!address) {
+      error = (
+        <T id="send.errors.invalidAddress" m="Please enter a valid address" />
+      );
+    }
+    try {
+      const validated = await validateAddress(address);
+      if (!validated.getIsValid()) {
+        error = (
+          <T id="send.errors.invalidAddress" m="Please enter a valid address" />
+        );
+      }
+      ref.data.error.address = error;
+      onUpdateOutput(ref);
+      onAttemptConstructTransaction();
+    } catch (err) {
+      return err;
+    }
+  };
+
+  const onValidateAmount = ({ atomValue, index }) => {
+    let error;
+    if (!atomValue || isNaN(atomValue)) {
+      error = (
+        <T id="send.errors.invalidAmount" m="Please enter a valid amount" />
+      );
+    }
+    if (atomValue <= 0) {
+      error = (
+        <T
+          id="send.errors.negativeAmount"
+          m="Please enter a valid amount (> 0)"
+        />
+      );
+    }
+    const ref = { ...outputs[index] };
+    ref.data.amount = atomValue;
+    ref.data.error.amount = error;
+    onUpdateOutput(ref);
+    onAttemptConstructTransaction();
+  };
+
+  const onChangeAccount = (account) => {
+    setAccount(account);
+    onAttemptConstructTransaction();
+  };
+
+  const onShowSendSelf = () => {
+    const newOutputs = [{ ...outputs[0], data: baseOutput().data }];
+    setIsSendSelf(true);
+    onSetOutputs(newOutputs);
+    onAttemptConstructTransaction();
+  };
+
+  const onShowSendOthers = () => {
+    const newOutputs = [{ ...outputs[0], data: baseOutput().data }];
+    setIsSendSelf(false);
+    onSetOutputs(newOutputs);
+    onAttemptConstructTransaction();
+  };
+
+  const onShowSendAll = () => {
+    const newOutputs = [{
+      ...outputs[0],
+      data: {
+        ...outputs[0].data,
+        amount: account.spendable
+      }
+    }];
+    setIsSendAll(true);
+    onSetOutputs(newOutputs);
+    onAttemptConstructTransaction();
+  };
+
+  const onKeyDown = (e) => {
+    if (e.keyCode === 13 && isValid()) {
+      setShowPassphraseModal(true);
+    }
+  };
+
+  const onHideSendAll = () => {
+    const newOutputs = [{
+      ...outputs[0],
+      data: {
+        ...outputs[0].data,
+        amount: null
+      }
+    }];
+    setIsSendAll(false);
+    onSetOutputs(newOutputs);
+    onAttemptConstructTransaction();
+  };
+
+  const hasError = useCallback(() => {
+    let hasError = false;
+    outputs.forEach((o) => {
```

Comment:
I'd use the piece of code below to improve readability
```javascript
const outputHasError = (o) =>!o.data.amount ||
        o.data.destination.length === 0 ||
        o.data.error.amount ||
        o.data.error.address
return outputs.some(outputHasError);
```

Reply:
liked the suggestion mate, committed